### ISSUE TITLE
Sexist examples used, please replace with something gender neutral

### ISSUE THREAD

**Issue by andreamazza89** [floere/picky#144]
Hi, thanks for sharing the gem.

I noticed this issue has already been discussed on #113 , but I am sad to see that the examples used in the docs are still of sexist nature. 

It would be very much appreciated if gender neutral examples were used instead.

Thanks 😄

**Comment by yvettecook** [comment]
Suggested improvement.

TL;DR: 🐱  

## Setting up Picky to use MongoDB with MongoMapper

Note by Florian: Sorry for the weird example (Discussed in: https://github.com/floere/picky/issues/113). Provided by the community, and I appreciate the work even though the example is very old-fashioned – please improve if you have the time.

For MongoDB awesomeness, just add the following files to your picky-server
application.

Add the mongomapper gem to your Gemfile

## Gemfile

```ruby
    gem 'mongo_mapper'
```


Run 

    bundle install


### lib/initializers/mongo.rb

```ruby
    settings = YAML.load(File.read(File.expand_path('../../config/mongo.yml', File.dirname(__FILE__))))["#{PICKY_ENVIRONMENT}"]
    MongoMapper.connection = Mongo::Connection.new(settings['host'], settings['port'])
    MongoMapper.database = settings['database']
```

### config/mongo.yml

```yaml
    development:
      adapter: mongodb
      database: fate_of_men_development
      host: localhost

    production:
      adapter: mongodb
      database: fate_of_men_production
      host: localhost

    test:
      adapter: mongodb
      database: fate_of_men_test
      host: localhost
```

## How to use embedded documents with picky server

There might be several ways to approach this problem, a really cool way is to use
a Delegator class. Let us assume, we have a model called man.
This man has one cat. The cat has a few wishes.

### lib/models/man.rb

```ruby
    class Man
      include MongoMapper::Document
      
      key :first_name,     String
      key :last_name,      String
      key :favourite_beer, String
      key :job,            String

      has_one :cat
    end
```

### lib/models/cat.rb

```ruby
    class Cat
      include MongoMapper::EmbeddedDocument

      key :name,      String
      
      belongs_to :man
      has_many :wishes      
    end
```

### lib/models/wish.rb

```ruby
    class Wish
      include MongoMapper::EmbeddedDocument
      
      key :type,     String
      key :cost,     Integer

      belongs_to :cat
    end
```

Ok, so how do we help our poor man to keep track of the wishes of his cat with picky power?
As mentioned above with a Delegator class

### lib/cat_delegator.rb

```ruby
    module CatDelegator
      def cat
        cat.name
      end

      def wish_type
        cat.wishes.map(&:type).join(" ") unless cat.wishes.blank?
      end
  
      def wish_cost
        cat.wishes.map(&:cost).join(" ") unless cat.wishes.blank?
      end
    end
```

Change your man class to look like this:

### lib/models/man.rb

```ruby
    require "cat_delegator"

    class man
        include MongoMapper::Document
        include CatDelegator

        ...
    end
```

Last step is to change the application.rb accordingly:

### app/application.rb

```ruby
    require "lib/models/man"
    require "lib/models/cat"
    require "lib/models/wish"

    class PickySearch < Application   
      ...

      men = Index::Memory.new(:men) do
        source Man.all

        key_format :to_s
        category :first_name,       partial: Partial::Substring.new(from: 1)
        category :last_name,        partial: Partial::Substring.new(from: 1)
        category :favourite_beer,	partial: Partial::Substring.new(from: 3)
        category :job,          	partial: Partial::Substring.new(from: 3)

        category :cat,           	partial: Partial::Substring.new(from: 1)
        category :wish_type,        partial: Partial::Substring.new(from: 3)
        category :wish_cost,        partial: Partial::Substring.new(from: 3)
      end
 
      complete_search = Search.new men
      route %r{^/men$} => complete_search
    end
```

Well, that's all folks.

**Comment by andreamazza89** [comment]
Amazing, thank @yvettecook

**Comment by yvettecook** [comment]
@floere: Proposed update for the 'old-fashioned' example in the docs. Just needs c+p over if you agree it's an improvement 👍

**Comment by floere** [comment]
@andreamazza89 Thanks for the pointer 😊 and
@yvettecook thanks so much for the improvement! 🎉 Feel free to simply edit the Wiki page so your contribution is properly attributed! I am very much in favor of the change 😊

**Comment by floere** [comment]
@yvettecook P.S: Also feel free to make it woman instead of man – there are enough examples with men in this world already 😉

**Comment by yvettecook** [comment]
@floere all done, balance in the universe restored.

![giphy](https://cloud.githubusercontent.com/assets/6195662/17771171/b10ba85a-6539-11e6-9590-8cfa4be1b302.gif)

**Comment by andreamazza89** [comment]
Awesome, thank @yvettecook!!

**Comment by floere** [comment]
@yvettecook Thanks so much, Yvette! 👏🚀❤️

**Comment by floere** [comment]
@andreamazza89 @yvettecook I am curious how this came about – how did you find it, and how did so many 👍🏽 happen so quickly? 😊
